pandas/core/indexes/base.py

```diff
@@ -695,13 +695,31 @@ def memory_usage(self, deep=False):
     # ops compat
     @deprecate_kwarg(old_arg_name='n', new_arg_name='repeats')
     def repeat(self, repeats, *args, **kwargs):
-        """
-        Repeat elements of an Index. Refer to `numpy.ndarray.repeat`
-        for more information about the `repeats` argument.
+        """Repeat elements of an Index.
```

Comment:
I think it should be useful to have an extended summary explaining why this method is useful. I mean, in which real cases this method could be used. I'm not sure myself, I don't think I've ever used it, but I think it should be very valuable to let users quickly understand whether this was the method they were looking for or not.

Also, the extended summary could explain how they are repeated. In the examples it's shown that they are repeated like `A, A, B, B` and not `A, B, A, B`. And I think it'd be nice to also have a bit more information about it.

Reply:
@datapythonista how does this look? Definitely contrived, but I don't really see this function being *that* useful. `Series.repeat` on the other hand...

Reply:
Hmm, as you say yourself it is rather contrived. 
If we don't find a good example, I would rather leave it out than adding in one that is a bit strange.

The main reason this exists is because of numpy compatibility (Index was previously a numpy array subclass). 
I think the method can be useful, but in very specific cases where you need something like that, too specific for the docstring. I think showing a generic example from which it is clear *what* it does (but without a *why*) is good enough in this case (like your first example).

Reply:
Sounds good. Will change and send another commit.

Reply:
@jorisvandenbossche, It's just a personal opinion, but if I ever end up in the doc of this method, I'd love to find something like exactly what you said, `The main reason this exists is because of numpy compatibility (Index was previously a numpy array subclass).`.

May be it's just me, but if after reading the documentation I still don't get what a function is for, and in which cases it should be useful for me, I think the documentation is not doing a good job. If a function still exists for legacy/compatibility reasons, I personally find it very useful to let the users know.

Reply:
> May be it's just me, but if after reading the documentation I still don't get what a function is for, and in which cases it should be useful for me, I think the documentation is not doing a good job. 

I think with the current example, it should be clear *what* it does, but I agree the *why* I would use it is less clear. 

> If a function still exists for legacy/compatibility reasons, I personally find it very useful to let the users know.

That's the orginal reason it is there, but nonetheless it has probably a use case (in contrast with some other methods/attributes like `Series.imag` which are *really* only there for historical reasons, but those are also not listed in api.rst). 
I know I already used it (or numpy's couterpart, not sure), but it was probably a little part of a complex analysis flow and can't remember exactly why I used it. And anyhow without the full context, it might not make much sense or be difficult to understand.

I understand your point, and it would indeed be good to include a good example illustrating a real-world use case, but then we need to come up with one ..

Reply:
How does this docstring look? Anything I need to change?